fuse_loss/CMakeLists.txt

```diff
@@ -1,36 +1,28 @@
-cmake_minimum_required(VERSION 3.0.2)
+cmake_minimum_required(VERSION 3.16)
 project(fuse_loss)
 
-set(build_depends
-  fuse_core
-  pluginlib
-  roscpp
-)
+# Default to C++17
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 17)
+  set(CXX_STANDARD_REQUIRED YES)
+endif()
 
-find_package(catkin REQUIRED COMPONENTS
-  ${build_depends}
-)
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
 
-find_package(Ceres REQUIRED)
+find_package(ament_cmake REQUIRED)
+find_package(fuse_core REQUIRED)
+find_package(pluginlib REQUIRED)
 
-catkin_package(
-  INCLUDE_DIRS
-    include
-  LIBRARIES
-    ${PROJECT_NAME}
-  CATKIN_DEPENDS
-    ${build_depends}
-  DEPENDS
-    CERES
-)
+find_package(Ceres REQUIRED)
 
 ###########
 ## Build ##
 ###########
 
-add_compile_options(-Wall -Werror)
-
-add_library(${PROJECT_NAME}
+## fuse_loss library
+add_library(${PROJECT_NAME} SHARED
```

Comment:
If you intentionally leave off `SHARED` then this library can be built statically by setting `-DBUILD_SHARED_LIBS=OFF` in CMake.

This is one of those weird CMake features/conventions, but  It looks like it was being used here before since `SHARED` wasn't specified. `ament_cmake_ros` documents that option when it's `find_package()'d`, so change `find_package(ament_cmake REQUIRED)` to `find_package(ament_cmake_ros REQUIRED)` above too.

Reply:
I'll also remove the SHARED on `fuse_core` in this PR too. It was added by me in a previous PR.